csharp/test/Ice/proxy/AllTests.cs

```diff
@@ -882,15 +883,15 @@ public static async Task<Test.MyClassPrx> allTests(global::Test.TestHelper helpe
                     inEncaps[4] = version.major;
                     inEncaps[5] = version.minor;
                     byte[] outEncaps;
-                    cl.ice_invoke("ice_ping", Ice.OperationMode.Normal, inEncaps,
-                                                                  out outEncaps);
+                    cl.ice_invoke("ice_ping", Ice.OperationMode.Normal, inEncaps, out outEncaps);
                     test(false);
                 }
                 catch (Ice.UnknownLocalException ex)
                 {
                     // TODO: remove UnsupportedEncodingException
                     test(
-                        ex.unknown.Contains("MarshalException") ||
+                        ex.unknown.Contains("::Ice::MarshalException") ||
+                        ex.unknown.Contains("Ice.MarshalException") ||
```

Comment:
It's a typo, will remove.

Reply:
Sorry, misread the code. We need `::Ice::MarshalException` + `Ice.MarshalException` for cross-testing. 

The previous `UnsupportedEncodingException` is just a lazy check for both versions. A downside is it can more easily match a string in the stack trace.